mantidproject/mantid
# Indirect Data Analysis - Keep open option for Multiple Input

**Description of work.**
This PR changes the `Cancel` and `Ok` buttons for the buttons `Close` and `Add` in the dialog window used to add workspaces in the Multiple Input tab of MSDFit, ConvFit, IqtFit and F(Q)Fit. 
This option is different to the previous options because clicking `Add` will add the data but the dialog window will not be closed (previously, clicking `Ok` would add the data and then close the dialog).
`Close` will just close the dialog window without adding the data selected.

**To test:**
1.`Interfaces`->`Indirect`->`Data Analysis`->Any of the above interfaces
2. Select the `Multiple Input` tab
3. Click `Add Workspace`
4. Load in a data file seen below and select the spectra you want to load
5. Click `Add`. 
Make sure the dialog stays open, and that data is added to the table seen in the Multiple Input tab
6. Load different data/select a different spectrum
7. Click `Add` and ensure this data has also been added to the table. 

Clicking `Close` should then close the window.

**Data Files**
**MSDFit:**
[osi104367-104371_graphite002_red_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2685812/osi104367-104371_graphite002_red_elwin_eq.zip)
[osi92762_graphite002_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2685814/osi92762_graphite002_elwin_eq.zip)
**IqtFit:**
[iris26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2685817/iris26176_graphite002_red.zip)
[irs26173_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2685828/irs26173_graphite002_red.zip)

**ConvFit:**
[iris26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2685819/iris26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2685825/irs26173_graphite002_res.zip)

**F(Q)Fit:**
[IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2685821/IN16B_125878_QLd_Result.zip)
[iris26174_graphite002_s0-17_Result.zip](https://github.com/mantidproject/mantid/files/2685822/iris26174_graphite002_s0-17_Result.zip)

Fixes #23813 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
